### PR TITLE
Scheduler: throw UnsupportedOperationException if appropriate

### LINKED ISSUE #44101: Scheduler: clarify what happens if scheduler is not started but method declared on Scheduler API is invoked
### Description

I think that all methods (except for `isRunning()`) should throw an `UnsupportedOperationException`.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by mkouba**
- also refactor the implementation and introduce BaseScheduler
- resolves #44101

### CODE REVIEW

**Review by manovotn — APPROVED** [review]
Added just one tiny comment with optional change.
I don't insist on it being modified before merging if you don't think it's needed.

File: extensions/scheduler/api/src/main/java/io/quarkus/scheduler/Scheduler.java [hunk 1]
```diff
@@ -17,7 +17,16 @@
 public interface Scheduler {
 
     /**
-     * Pause the scheduler. No triggers are fired.
+     * By default, the scheduler is not started unless a {@link Scheduled} business method is discovered.
```
[the remark this answers is not in the document]
Comment (mkouba) [reply extensions/scheduler/api/src/main/java/io/quarkus/scheduler/Scheduler.java:20]: Good point.

[the remark this answers is not in the document]
Comment (mkouba) [reply extensions/scheduler/api/src/main/java/io/quarkus/scheduler/Scheduler.java:20]: What about this?

```suggestion
     * By default, the scheduler is not started unless a {@link Scheduled} business method is discovered. However, it is
     * possible to set the {@code forced} start mode with the {@code quarkus.scheduler.start-mode} configuration property. In
     * this case, the scheduler will be started even if no scheduled business methods are found.
```

[the remark this answers is not in the document]
Comment (manovotn) [reply extensions/scheduler/api/src/main/java/io/quarkus/scheduler/Scheduler.java:20]: Yes, that looks good :+1:

**Pull request merged**